src/builders/CommandStep.ts

```diff
@@ -65,19 +72,70 @@ export class CommandStep implements StepBuilder, KeyBuilder, LabelBuilder, Condi
     this.#plugins.push(plugin)
     return this
   }
+  
+  parallelism(parallelism: number): this {
+    this.#parallelism = parallelism
+    return this
+  }
+  
+  env(env: Record<string, string | number>): this {
+    this.#env = env
+    return this
+  }
 
-  build(): CommandStepSchema {
-    if (!this.#command) {
-      throw new Error('CommandStep requires a command.')
+  addEnv(key: string, value: string | number): this {
+    if (!this.#env) {
+      this.#env = {}
+    }
+    this.#env[key] = value
+    return this
+  }
+
+  concurrency(jobs?: number, group?: string): this {
+    this.#concurrency = jobs
+    this.#concurrency_group = group
+    return this
+  }
+
+  softFail(fail: boolean = true): this {
+    this.#soft_fail = fail
+    return this
+  }
+
+  timeout(minutes: number): this {
+    this.#timeout_in_minutes = minutes
+    return this
+  }
+
+  artifactPath(path: string): this {
+    if (!this.#artifact_paths) {
+      this.#artifact_paths = []
     }
+    this.#artifact_paths.push(path)
+    return this
+  }
+
+  build(): CommandStepSchema {
+    const commandKey = this.#command && Array.isArray(this.#command) && this.#command.length > 1 ? 'commands' : 'command'
     const object: CommandStepSchema = {
-      command: this.#command,
+      ...{ [commandKey]: this.#command },
       ...this.#keyHelper.build(),
       ...this.#labelHelper.build(),
       ...this.#conditionHelper.build(),
       ...this.#branchesHelper.build(),
       ...this.#dependenciesHelper.build(),
       ...this.#skipHelper.build(),
+      ...(this.#parallelism ? { parallelism: this.#parallelism } : {}),
+      ...(this.#concurrency ? {concurrency: this.#concurrency} : {}),
```

Comment:
☝️ for these numeric values, is `0` a valid value we need to handle?

Reply:
🤦

Reply:
Actually, both provide behaviour that I can't imagine would ever be used intentionally:
- `parallelism: 0` just removes the step
- `concurrency: 0` just causes the step to hang

Reply:
Added errors for both cases

Reply:
Twas just checking cause `0 == false` and would result in `{}` when the values are `0`